File: src/dataset/javascript/acronym_exercism-SLIB53/gemini/GeminiFlash_acronym_v3.js
export const parse = phrase => {
  let result = "";
  const upperPhrase = phrase.toUpperCase();
  for (let i = 0; i < upperPhrase.length; i++) {
    const char = upperPhrase[i];
    if (char >= 'A' && char <= 'Z') {
      result += char;
      while (i + 1 < upperPhrase.length && upperPhrase[i + 1] >= 'A' && upperPhrase[i + 1] <= 'Z') {
        i++;
      }
    } else if (char === "'") {
      continue;
    }
  }
  return result;
};

File: src/dataset/Java/acronym_exercism-awsaf174/gemini/GeminiFlash_Acronym_v3.java
class Acronym {

  private final String acronym;

  Acronym(String phrase) {
    StringBuilder result = new StringBuilder();
    phrase = phrase.replace('-', ' ');
    phrase = phrase.replaceAll("[^A-Za-z0-9\\s]", "");
    phrase = phrase.replaceAll("\\s+", " ");
    phrase = phrase.trim();

    if (!phrase.isEmpty()) {
      result.append(Character.toUpperCase(phrase.charAt(0)));

      for (int i = 1; i < phrase.length(); i++) {
        if (Character.isWhitespace(phrase.charAt(i))) {
          if (i + 1 < phrase.length()) {
            result.append(Character.toUpperCase(phrase.charAt(i + 1)));
            i++;
          }
        }
      }
    }
    this.acronym = result.toString();
  }

  String get() {
    return acronym;
  }
}